ns1/ipam.py

```diff
@@ -0,0 +1,446 @@
+
+    def update(self, callback=None, errback=None, **kwargs):
+        """
+        Update VRF configuration. Pass a list of keywords and their values to
+        update. For the list of keywords available for zone configuration, see
+        :attr:`ns1.rest.ipam.Vrfs.INT_FIELDS` and
+        :attr:`ns1.rest.ipam.Vrfs.PASSTHRU_FIELDS`
+        """
+        if not self.data:
+            raise VrfException('VRF not loaded')
+
+        def success(result, *args):
+            self.data = result
+            self.id = result['id']
+            self.name = result['name']
+            self.route_table = result['rt']
+            self.report = self._rest.report(self.id)
+            if callback:
+                return callback(self)
+            else:
+                return self
+
+        return self._rest.update(self.id, callback=success, errback=errback,
+                                 **kwargs)
+
+    def create(self, callback=None, errback=None, **kwargs):
+        """
+        Create a new VRF. Pass a list of keywords and their values to
+        configure. For the list of keywords available for vrf configuration,
+        see :attr:`ns1.rest.ipam.Vrfs.INT_FIELDS` and
+        :attr:`ns1.rest.ipam.Vrfs.PASSTHRU_FIELDS`
+        """
+        if self.data:
+            raise VrfException('VRF already loaded')
+
+        def success(result, *args):
+            self.data = result
+            self.id = result['id']
+            self.name = result['name']
+            self.route_table = result['rt']
+            self.report = self._rest.report(self.id)
+            if callback:
+                return callback(self)
+            else:
+                return self
+
+        return self._rest.create(rt=self.route_table, name=self.name, callback=success, errback=errback, **kwargs)
+
+    def new_address(self, prefix, type, callback=None, errback=None):
+        """
+        Create a new address space in this VRF
+        :param str prefix: The CIDR prefix of the address to add
+        :param str type: planned, assignment, host
+        :param callback:
+        :param errback:
+        :return:
+        """
+        if not self.data:
+            raise VrfException('VRF not loaded')
+
+        return Address(self.config, prefix, type, self).create()
+
+
+class Address(object):
+
+    def __init__(self, config, prefix=None, type=None, vrf=None, id=None):
+        """
+        Create a new high level Address object
+
+        :param ns1.config.Config config: config object
+        :param str prefix: cidr prefix
+        :param str type: planned, assignment, host
+        :param Vrf vrf: VRF Object the address will be part of
+        """
+        self._rest = Addresses(config)
+        self.config = config
+        self.prefix = prefix
+        self.type = type
+        self.vrf = vrf
+        self.id = id
+        self.children = []
+        self.report = {}
+        self.data = None
```

Comment:
This doesn't yet address the optional scope group association; it's captured in the API by `scope_group_id`, so maybe that means we should pass in an SG analogous to vrf?

Reply:
Makes sense, just checked in a change to address this